src/Cache/Engine/XcacheEngine.php

```diff
@@ -80,7 +80,7 @@ public function write($key, $value)
         $duration = $this->_config['duration'];
         $expires = time() + $duration;
         xcache_set($key . '_expires', $expires, $duration);
-        return xcache_set($key, $value, $duration);
+        return xcache_set($key, serialize($value), $duration);
```

Comment:
You can inject PHP (object code) which would blindly be executed by unserialize().